lib/utils.js

```diff
@@ -85,11 +88,11 @@ module.exports.parseSamlRequest = function(req, samlRequest, type, options, call
   if (!samlRequest) return callback();
 
   try {
-    var input = new Buffer(samlRequest, 'base64');
+    const input = new Buffer(samlRequest, 'base64');
 
     if (input[0] === 60) {  // open tag
       // content is just encoded, not zipped
-      var xml = new xmldom.DOMParser().parseFromString(input.toString());
+      const xml = new xmldom.DOMParser().parseFromString(input.toString());
       if (!xml || !xml.documentElement) {
```

Comment:
It looks like this bit of code is the same in the else statement the only difference is that we're unzipping it, I wonder if this bit of code could be abstracted into a function which is passed the result of `.toString()`.
```javascript
const xml = new xmldom.DOMParser().parseFromString(input);

if (!xml || !xml.documentElement) {
   return callback(new Error('Invalid SAML Request'));
}

configureSigningCredentials(xml, function (err, shouldValidate) {
   if (!shouldValidate) {
          return callback(null, xml);
   }
   checkSignature(xml, callback);
});
```
I think doing this should mean we can avoid setting `xml = null` in the next block as well.

Reply:
Well spotted. I usually "lift" the optional computation into a local const to keep the computation linear.

```
const input = new Buffer(samlRequest, 'base64');
const inflateIfNeeded = input[0] === 60 ? (cb) => setImmediate(cb, null, input) : zlib.inflateRaw;

inflateIfNeeded(input, function(err, inflated) {
  // shared logic
});
```

Reply:
Thats nice, I think I prefer that way!

Reply:
This is a great spot and change, but I'm apprehensive to apply it in this particular PR & ticket.
There's a few different areas in the utils file (and across the lib) that could do with a refactor (misspelled opts, etc.).  I'll place a ticket on the backlog that references this suggestion and a few other proposals.

Reply:
Yeah totally, that sounds sensible to me.